twitchio/ext/sounds/queuemanager.py

```diff
@@ -0,0 +1,141 @@
+        self.is_playing = False
+        self.queue.task_done()
+
+    async def add_audio(self, sound_path: str) -> None:
+        """
+        Adds a new audio file to the queue.
+
+        Args:
+            sound_path (str): Path of the audio file to add to the queue.
+        """
+        await asyncio.sleep(0.1)
+        await self.queue.put(sound_path)
+
+    async def play_next(self) -> None:
+        """
+        Plays the next audio file in the queue if the queue is not empty and not paused.
+        Sets the is_playing flag, retrieves the next audio file from the queue, and plays it.
+        If repeat_queue is True, adds the current audio file back to the queue after playing.
+        """
+        await asyncio.sleep(0.1)
+        if not self.queue.empty() and not self.queue_paused:
+            self.is_playing = True
+            sound_path = await self.queue.get()
+            self.current_sound = sound_path
+            sound = sounds.Sound(source=sound_path)
+            self.player.play(sound)
+            if self.repeat_queue:
+                await self.queue.put(self.current_sound)
+
+    async def skip_audio(self) -> None:
+        """
+        Stops the currently playing audio file if there is one.
+        """
+        await asyncio.sleep(0.1)
+        if self.is_playing:
+            self.player.stop()
+            self.is_playing = False
+
+    async def stop_audio(self) -> None:
+        """
+        Stops the currently playing audio file.
+        Resets the playing flag but leaves the queue intact.
+        """
+        await asyncio.sleep(0.1)
+        if self.is_playing:
+            self.player.stop()
+            self.is_playing = False
+
+    async def pause_audio(self) -> None:
+        """
+        Pauses the currently playing audio file.
+        """
+        await asyncio.sleep(0.1)
+        self.player.pause()
+
+    async def resume_audio(self) -> None:
+        """
+        Resumes the currently paused audio file.
+        """
+        await asyncio.sleep(0.1)
+        self.player.resume()
+
+    async def clear_queue(self) -> None:
+        """
+        Clears all audio files from the queue.
+        """
+        await asyncio.sleep(0.1)
+        while not self.queue.empty():
+            await self.queue.get()
+            self.queue.task_done()
+
+    async def pause_queue(self) -> None:
+        """
+        Pauses the processing of the queue.
+        """
+        await asyncio.sleep(0.1)
+        self.queue_paused = True
+
+    async def resume_queue(self) -> None:
+        """
+        Resumes the processing of the queue.
+        """
+        await asyncio.sleep(0.1)
+        self.queue_paused = False
+
+    async def get_queue_contents(self) -> list:
+        """
+        Retrieves the current contents of the queue.
+
+        Returns:
+            list: List of paths of audio files in the queue.
+        """
+        await asyncio.sleep(0.1)
+        return list(self.queue._queue)
```

Comment:
This isn't exposed by asyncio.Queue. 
Depending on Mysty's input, you may need to create a private attribute in your class AudioQueueManager.
e.g.
```
self._all_items = [] # Or make it a deque.

    async def enqueue(self, item):
        await self.queue.put(item)
        self._all_items.append(item)  # Mirror the item in _all_items

    def get_queue_contents(self):
        return list(self._all_items)
```

Although for 2.x the current implementation may be fine and we'll rewrite the whole extension for 3.x

Reply:
Mysty said its fine the way it is currently, thanks! I see your point though, I hadnt thought about that.